main.cpp

```diff
@@ -1,35 +1,50 @@
 /* 基于智能指针实现双向链表 */
 #include <cstdio>
 #include <memory>
+#include <iostream>
 
 struct Node {
     // 这两个指针会造成什么问题？请修复
-    std::shared_ptr<Node> next;
-    std::shared_ptr<Node> prev;
+    std::unique_ptr<Node> next = nullptr;
+    Node* prev = nullptr;
     // 如果能改成 unique_ptr 就更好了!
 
     int value;
 
     // 这个构造函数有什么可以改进的？
-    Node(int val) {
-        value = val;
+    Node(int val) :value(val) {
```

Comment:
```suggestion
    Node(int val) : value(val) {
```
通常所说的代码规范会在符号前后会有空格。